src/plugins/intel_cpu/tests/functional/single_layer_tests/instances/x64/multinomial.cpp

```diff
@@ -0,0 +1,120 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "single_layer_tests/classes/multinomial.hpp"
+
+#include <openvino/core/type/element_type.hpp>
+#include <openvino/runtime/tensor.hpp>
+
+using namespace CPUTestUtils;
+using namespace ov::test;
+
+namespace CPULayerTestsDefinitions {
+namespace Multinomial {
+
+const int GLOBAL_SEED = 1;
+const int OP_SEED = 2;
+
+const std::vector<float> probs_4x4_f32 =
+    {0.00001f, 0.001f, 0.1f, 10.0f, 0.001f, 0.00001f, 10.0f, 0.1f, 0.1f, 10.0f, 0.00001f, 0.001f, 10.0f, 0.1f, 0.001f, 0.00001f};
+
+const std::vector<ov::float16> probs_2x3_f16 =
+    {(ov::float16)0.001f, (ov::float16)0.1f, (ov::float16)10.0f, (ov::float16)10.0f, (ov::float16)0.001f, (ov::float16)0.1f};
+
+const std::vector<ov::bfloat16> probs_1x3_bf16 = {(ov::bfloat16)0.001f, (ov::bfloat16)0.1f, (ov::bfloat16)10.0f};
+
+const std::vector<float> probs_4x4_f32_log =
+    {3.0f, 6.0f, 10.0f, 0.0f, 3.0f, 0.0f, 10.0f, 6.0f, 6.0f, 10.0f, 0.0f, 3.0f, 10.0f, 6.0f, 3.0f, 0.0f};
+
+const std::vector<ov::float16> probs_2x3_f16_log =
+    {(ov::float16)3.0f, (ov::float16)6.0f, (ov::float16)10.0f, (ov::float16)10.0f, (ov::float16)3.0f, (ov::float16)6.0f};
+
+const std::vector<ov::bfloat16> probs_1x3_bf16_log = {(ov::bfloat16)3.0f, (ov::bfloat16)6.0f, (ov::bfloat16)10.0f};
+
+const std::vector<int> num_samples_scalar_i32 = {1};
+const std::vector<int64_t> num_samples_1x1_i64 = {2};
+const std::vector<int64_t> num_samples_scalar_i64 = {3};
+
+const std::vector<ov::Tensor> probs = {ov::Tensor(ov::element::f32, {4, 4}, (void*)probs_4x4_f32.data()),
+                                       ov::Tensor(ov::element::f16, {2, 3}, (void*)probs_2x3_f16.data()),
+                                       ov::Tensor(ov::element::bf16, {1, 3}, (void*)probs_1x3_bf16.data())};
+
+const std::vector<ov::Tensor> probs_log = {ov::Tensor(ov::element::f32, {4, 4}, (void*)probs_4x4_f32_log.data()),
+                                           ov::Tensor(ov::element::f16, {2, 3}, (void*)probs_2x3_f16_log.data()),
+                                           ov::Tensor(ov::element::bf16, {1, 3}, (void*)probs_1x3_bf16_log.data())};
+
+const std::vector<ov::Tensor> num_samples = {ov::Tensor(ov::element::i32, {}, (void*)num_samples_scalar_i32.data()),
+                                             ov::Tensor(ov::element::i64, {1}, (void*)num_samples_1x1_i64.data()),
+                                             ov::Tensor(ov::element::i64, {}, (void*)num_samples_scalar_i64.data())};
+
+const std::vector<ov::test::ElementType> convert_type = {ov::test::ElementType::i32};
```

Comment:
Based on specification doc for Multinomial, both i32 and i64 are allowed, shouldn't we support and test it?

Reply:
Right, added the support with the new API

Reply:
@nshchego Just to verify, is the i64 support available in CPU plugin?